numpy/lib/tests/test_io.py

```diff
@@ -1068,6 +1068,52 @@ def test_binary_load(self):
             x = [b'5,6,7,\xc3\x95scarscar', b'15,2,3,hello', b'20,2,3,\xc3\x95scar']
             assert_array_equal(x, np.array(x, dtype="S"))
 
+    def test_max_rows(self):
+        c = TextIO()
+        c.write('1,2,3,5\n4,5,7,8\n2,1,4,5')
+        c.seek(0)
+        x = np.loadtxt(c, dtype=int, delimiter=',',
+                       max_rows=1)
+        a = np.array([1, 2, 3, 5], int)
+        assert_array_equal(x, a)
+
+        c = TextIO()
```

Comment:
This would be better as a bunch of separate tests, each starting at the `c = TextIO()` line

Reply:
Done. I split them into few test cases with some (hopefully) descriptive names